src/components/TextInput.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { ReactElement } from 'react';
+import {
+    FormControl as FormControlMUI,
+    InputLabel as InputLabelMUI,
+    Input as InputMUI,
+    FilledInput as FilledInputMUI,
+    SxProps,
+    Theme,
+} from '@mui/material';
+
+interface TextInputProps {
+    /**
+     * HTML element id
+     */
+    id?: string;
+    /**
+     * HTML input element type, defaults to text
```

Comment:
nit:

```suggestion
     * HTML input element type, defaults to `text`
```
Same can be done on line 33